web-components/src/components/list-v2/list-v2.ts

```diff
@@ -0,0 +1,110 @@
+import { customElement, html, LitElement, property, queryAssignedNodes } from "lit-element";
+import style from "./scss/module.scss";
+import { ListItemV2 } from "./list-item-v2";
+import { Key } from "@/constants";
+
+export namespace ListV2 {
+  export type Gap = "none" | "sm" | "md" | "lg";
+  export type Orientation = "vertical" | "horizontal";
+
+  @customElement("md-list-v2")
+  export class ELEMENT extends LitElement {
+    @property({ type: String }) gap: Gap = "none";
+    @property({ type: String }) orientation: Orientation = "vertical";
+
+    @queryAssignedNodes() assignedNodes!: NodeList;
+
+    connectedCallback(): void {
+      super.connectedCallback();
+      this.setAttribute("role", "list");
+      this.addEventListener("keydown", this.handleKeyDown);
+    }
+
+    render() {
+      return html` <slot @click=${this.handleMouseClick}></slot>`;
+    }
+
+    private get listItems() {
+      return Array.from(this.assignedNodes).filter((node) => node instanceof ListItemV2.ELEMENT);
+    }
+
+    private handleMouseClick(event: MouseEvent) {
+      const newIndex = this.getCurrentIndex(event.target);
+      this.resetTabIndexAndSetActiveTabIndex(newIndex);
+    }
+
+    /**
+     * Calculates a new index based on the pressed keyboard key.
+     * Supports navigation keys for moving focus within a list.
+     * @param key - The key that was pressed.
+     * @param currentIndex - The current index of the focused list item.
+     * @param wrappedDivsCount - The total number of list items.
+     * @returns The new index to focus on, or undefined if the key is not supported.
+     */
+    private getNewIndexBasedOnKey(key: string, currentIndex: number, wrappedDivsCount: number): number | undefined {
+      switch (key) {
+        case Key.ArrowDown:
+          return (currentIndex + 1) % wrappedDivsCount;
+        case Key.ArrowUp:
+          return (currentIndex - 1 + wrappedDivsCount) % wrappedDivsCount;
+        case Key.Home:
+          return 0;
+        case Key.End:
+          return wrappedDivsCount - 1;
+        default:
+          return undefined;
+      }
+    }
+
+    /**
+     * Handles the keydown event on the list element.
+     * If the key is 'ArrowUp' or 'ArrowDown', it focuses to the previous or next list item
+     * and sets the active tabindex of the list item.
+     * Prevents the default event behavior.
+     * @param event - The keyboard event.
+     */
+    private handleKeyDown(event: KeyboardEvent): void {
+      const currentIndex = this.getCurrentIndex(event.target);
+      const newIndex = this.getNewIndexBasedOnKey(event.key, currentIndex, this.listItems.length);
```

Comment:
When working with the shadow dom you can use this to see what is clicked on

const path = event.composedPath();
const originalTarget = path[0] as HTMLElement;